File: cpp/test/sorting/quicksort/quick_sort_test.cpp
//
// Created by nisha on 6/2/2020.
//

#include <gtest/gtest.h>
#include "../../../main/sorting/quicksort/quick_sort.h"

TEST(Sorting, QuickSortTest) {
  std::vector<int> arr = my_utils::createRandVector(10000, 1, 1000000);
  std::vector<int> copy = arr;
  
  std::sort(copy.begin(), copy.end());
  heap_sort(arr);
  ASSERT_EQ(copy, arr);
}

File: java/src/test/java/sorting/QuickSortTest.java
package sorting;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import sorting.quicksort.QuickSort;

public class QuickSortTest {

  private int[] arr;

  @BeforeEach
  public void setup() {
    arr = new int[] {5, 3, 2, 1, 5};
  }

  @Test
  public void test() {
    QuickSort.sort(arr);
    assertArrayEquals(new int[] {1, 2, 3, 5, 5}, arr);
  }
}
